vault/token_store.go

```diff
@@ -1396,13 +1447,36 @@ func (ts *TokenStore) tokenStoreRoleCreateUpdate(
 		entry.Period = time.Second * time.Duration(data.Get("period").(int))
 	}
 
+	var resp *logical.Response
+
+	explicitMaxTTLInt, ok := data.GetOk("explicit_max_ttl")
+	if ok {
+		entry.ExplicitMaxTTL = time.Second * time.Duration(explicitMaxTTLInt.(int))
+	} else if req.Operation == logical.CreateOperation {
```

Comment:
In general, for all the fields of all paths that distinguishes between CreateOperation and UpdateOperation, we can have this check of `else if req.Operation == logical.CreateOperation` only if the field under question has a default value that we desire to set. Otherwise we can skip this part. Thoughts?